ipld/unixfs/io/directory.go

```diff
@@ -231,12 +395,20 @@ func (d *BasicDirectory) addLinkChild(ctx context.Context, name string, link *ip
 	if err != nil && err != os.ErrNotExist {
 		return err
 	}
+	if err == nil { // existed
+		d.totalLinks--
+	}
```

Comment:
```suggestion
	if err != nil {
		if errors.Is(err, os.ErrNotExist) {
			return err
		}
	} else { // existed
		d.totalLinks--
	}
```

Reply:
It should be `!errors.Is(err, os.ErrNotExist)` but ok.